### PR TITLE
WanDB logging

### PR DESCRIPTION
**Pull request by Anustup900**
This PR contains WanDB logging

### CODE REVIEW

**Comment by Anustup900** [comment]
Need to add `WANDB_API_KEY` in `.env` pre-depolyment

File: predict.py [hunk 1]
```diff
@@ -45,17 +49,18 @@ def setup(self) -> None:
 
     def predict(
             self,
+            product_id : str = Input(description="Please enter Product ID here", default=None),
             image: Path = Input(description="Input image"),
             mask: Path = Input(description="Mask image", default=None),
             trimap: Path = Input(description="Trimap image", default=None),
     ) -> Output:
         # if there's no mask/trimap, return an error
+        initialize_wandb(product_id)
```
Comment (iamargentum) [inline predict.py:58]: move this to `setup` function

**Review by iamargentum — APPROVED** [review]
looks good
added one comment for refactor

(for me) TODO: Add `serverless_function` decorator over wandb and segmentation functions

File: utils.py [hunk 2]
```diff
@@ -275,6 +278,31 @@ def extra_edge_removal_from_matte_output(matte_image, output_path):
         raise_sentry_error(e)
         return {"success": False, "error": f"Edge removal failed due to :{e}"}
 
+def overlay_final_mask_in_black_background(image_path, mask_path, output_image_path):
```
Comment (iamargentum) [inline utils.py:281]: let's also incorporate the practice of writing docstrings, references and more comments that describe the code and motivation behind it
will be easier for the next person to understand

Comment (Anustup900) [reply utils.py:281]: sure will be adding this for now adding a `docstring `with this function

File: predict.py (hunk 1 above)
Comment (Anustup900) [reply predict.py:58]: Added this in `TODO `of Asana

**Pull request merged**